pyxrf/model/scan_metadata.py

```diff
@@ -0,0 +1,213 @@
+import re
+
+
+class ScanMetadataBase:
+    """
+    This is a base class. Instances of classes should be created from
+    a child class, which has meaningful implementation of the ``_gen_default_descriptions``
+    and ``_gen_print_order`` functions.
+    """
+
+    def __init__(self):
+
+        # This dictionary contains key-value pairs that represent parameter values
+        self._values = {}
+
+        # The following dictionary contains key-description pairs, that represent
+        #   user-friendly description of each key, that may be used for printing.
+        #   It is not assumed that there is one-to-one match for the keys
+        #   in 'values' and 'key_description' dictionaries. The 'key_descriptions'
+        #   should contain comprehensive list of keys. It is also assumed that
+        #   some keys in 'values' dictionary may not have matching descriptions.
+        #   'key_descriptions' dictionary set at the different place in the program.
+        #   In principle the descriptions may not be set at all.
+        self.descriptions = self._gen_default_descriptions()
+
+        self._print_order = self._gen_default_print_order()
+
+    def _gen_default_print_order(self):
+        """
+        This function must have meaningful implementation in the child class
+        """
+        return []
+
+    def _gen_default_descriptions(self):
+        """
+        This function must have meaningful implementation in the child class
+        """
+        return {}
+
+    # The following methods override standard operators to enable direct access
+    #   to ``self.values`` dictionary
+
+    def __getitem__(self, key):
+        return self._values[key]
+
+    def __setitem__(self, key, value):
+        self._values[key] = value
+
+    def __delitem__(self, key):
+        del self._values[key]
+
+    def __contains__(self, key):
+        return key in self._values
+
+    def __iter__(self):
+        return self._values.__iter__()
```

Comment:
I was thinking of subclassing from the dict, but it doesn't seem reasonable, since the class is not intended to be a dictionary and it is not clear at this point how much extra functionality will be added to it. Right now the metadata is stored as a separate dictionary ``_values``, which is independent of the rest of the class. In the future, metadata may be stored in a different container. It all depends on the direction in which future work on standardization of metadata goes. The implementation of access methods is now trivial, but it may change in the future if access to metadata is more complex.

Reply:
Sounds good.